src/main/java/betterquesting/client/gui2/GuiQuestLines.java

```diff
@@ -518,4 +526,77 @@ private void refreshClaimAll()
         claimAll.setIcon(PresetIcon.ICON_CHEST_ALL.getTexture(), new GuiColorStatic(0xFF444444), 0);
         claimAll.setActive(false);
     }
+
+    private void openSearch(PanelButton panelButton) {
+        this.resetCanvas();
+        CanvasTextured cvBackground = new CanvasTextured(new GuiTransform(GuiAlign.FULL_BOX, new GuiPadding(0, 0, 0, 0), 0), PresetTexture.PANEL_MAIN.getTexture());
+        this.addPanel(cvBackground);
+        CanvasEmpty cvRight = new CanvasEmpty(new GuiTransform(GuiAlign.FULL_BOX, new GuiPadding(8, 8, 8, 8), 0));
+        cvBackground.addPanel(cvRight);
+
+        PanelButton btnExit = new PanelButton(new GuiTransform(GuiAlign.BOTTOM_CENTER, -100, -16, 200, 16, 0), 0, QuestTranslation.translate("gui.back"));
+        btnExit.setClickAction((b) -> this.initPanel());
+        cvRight.addPanel(btnExit);
+
+        PanelTextBox txtDb = new PanelTextBox(new GuiTransform(GuiAlign.TOP_EDGE, new GuiPadding(0, 0, 0, -16), 0), QuestTranslation.translate("betterquesting.gui.search")).setAlignment(1).setColor(PresetColor.TEXT_MAIN.getColor());
+        cvRight.addPanel(txtDb);
+
+        PanelTextField<String> searchBox = new PanelTextField<>(new GuiTransform(GuiAlign.TOP_EDGE, new GuiPadding(0, 16, 8, -32), 0), "", FieldFilterString.INSTANCE);
+        searchBox.setWatermark("Search...");
+        cvRight.addPanel(searchBox);
+
+        GuiQuestLines thisGui = this;
+        UUID playerUUID = QuestingAPI.getQuestingUUID(mc.thePlayer);
+        CanvasQuestSearch canvasQuestSearch = new CanvasQuestSearch(new GuiTransform(GuiAlign.FULL_BOX, new GuiPadding(0, 32, 8, 24), 0)) {
+            @Override
+            protected boolean addResult(QuestSearchEntry entry, int index, int cachedWidth) {
```

Comment:
please do not use inline functions.

Reply:
What should I do then?

Reply:
So currently youre using an anonymous class in which you override a single function, it would be cleaner, if you would just make a new class file and use that here, instead of this approach